From patchwork Fri Jan 27 07:21:33 2023
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Jagan Teki <jagan@edgeble.ai>
X-Patchwork-Id: 1732735
X-Patchwork-Delegate: trini@ti.com
Return-Path: <u-boot-bounces@lists.denx.de>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@legolas.ozlabs.org
Authentication-Results: legolas.ozlabs.org;
 spf=pass (sender SPF authorized) smtp.mailfrom=lists.denx.de
 (client-ip=2a01:238:438b:c500:173d:9f52:ddab:ee01; helo=phobos.denx.de;
 envelope-from=u-boot-bounces@lists.denx.de; receiver=<UNKNOWN>)
Authentication-Results: legolas.ozlabs.org;
	dkim=pass (2048-bit key;
 unprotected) header.d=edgeble-ai.20210112.gappssmtp.com
 header.i=@edgeble-ai.20210112.gappssmtp.com header.a=rsa-sha256
 header.s=20210112 header.b=bHCu3uqG;
	dkim-atps=neutral
Received: from phobos.denx.de (phobos.denx.de
 [IPv6:2a01:238:438b:c500:173d:9f52:ddab:ee01])
	(using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
	 key-exchange X25519 server-signature ECDSA (P-384))
	(No client certificate requested)
	by legolas.ozlabs.org (Postfix) with ESMTPS id 4P38CX1yz3z23hC
	for <incoming@patchwork.ozlabs.org>; Fri, 27 Jan 2023 18:22:00 +1100 (AEDT)
Received: from h2850616.stratoserver.net (localhost [IPv6:::1])
	by phobos.denx.de (Postfix) with ESMTP id 17203855D3;
	Fri, 27 Jan 2023 08:21:55 +0100 (CET)
Authentication-Results: phobos.denx.de;
 dmarc=none (p=none dis=none) header.from=edgeble.ai
Authentication-Results: phobos.denx.de;
 spf=pass smtp.mailfrom=u-boot-bounces@lists.denx.de
Authentication-Results: phobos.denx.de;
	dkim=pass (2048-bit key;
 unprotected) header.d=edgeble-ai.20210112.gappssmtp.com
 header.i=@edgeble-ai.20210112.gappssmtp.com header.b="bHCu3uqG";
	dkim-atps=neutral
Received: by phobos.denx.de (Postfix, from userid 109)
 id 8238584B26; Fri, 27 Jan 2023 08:21:52 +0100 (CET)
X-Spam-Checker-Version: SpamAssassin 3.4.2 (2018-09-13) on phobos.denx.de
X-Spam-Level: 
X-Spam-Status: No, score=-1.9 required=5.0 tests=BAYES_00,DKIM_SIGNED,
 DKIM_VALID,SPF_HELO_NONE,SPF_NONE autolearn=ham autolearn_force=no
 version=3.4.2
Received: from mail-pf1-x42c.google.com (mail-pf1-x42c.google.com
 [IPv6:2607:f8b0:4864:20::42c])
 (using TLSv1.3 with cipher TLS_AES_128_GCM_SHA256 (128/128 bits))
 (No client certificate requested)
 by phobos.denx.de (Postfix) with ESMTPS id EBC9F84B26
 for <u-boot@lists.denx.de>; Fri, 27 Jan 2023 08:21:49 +0100 (CET)
Authentication-Results: phobos.denx.de;
 dmarc=none (p=none dis=none) header.from=edgeble.ai
Authentication-Results: phobos.denx.de;
 spf=none smtp.mailfrom=jagan@edgeble.ai
Received: by mail-pf1-x42c.google.com with SMTP id z31so2706937pfw.4
 for <u-boot@lists.denx.de>; Thu, 26 Jan 2023 23:21:49 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=edgeble-ai.20210112.gappssmtp.com; s=20210112;
 h=content-transfer-encoding:mime-version:message-id:date:subject:cc
 :to:from:from:to:cc:subject:date:message-id:reply-to;
 bh=jO5FfbyvXE8BQ34WprZvwW/bJZ/LqqpxrYklmSWOloM=;
 b=bHCu3uqGPBN2bve4LkDTOWrXw4hCt4hSgIS0hSiMQHkSaeTwA7hUU5PH/GCt9tcD4V
 DvBfawwAEKWlsERPkLIcy+fC9LTFluiMSgT3qBIeuJtUMjB/Dry62XFxZmgoWAD+gtIq
 evUJ4Qz3oHNIvdBNdrd7WGQ2q2tqPftVP/p/TXXfH7JxHQjLjZqJZBroJwJ3+uDPzn5E
 04+J9pgHYe7vFQ+CjwEqs7yAfyQsoqxBYeiMW1VE6gWqOZMWHb7/p8FP+SZhY8CHe4Ck
 eP0lt4vvrSiQdSnBlYSkeONVRZn/7KOmYmzJ9ZEp/UgAH5YQO2lFfP+JkzPzBEhS5zHU
 mKtw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20210112;
 h=content-transfer-encoding:mime-version:message-id:date:subject:cc
 :to:from:x-gm-message-state:from:to:cc:subject:date:message-id
 :reply-to;
 bh=jO5FfbyvXE8BQ34WprZvwW/bJZ/LqqpxrYklmSWOloM=;
 b=d4rnUnSBiBXsuojUhgMtyelI+UtyOFGMIB6Be/aPybUpaWCJm80DofP37e4XP3kiSm
 /pUCMqnoXIPHNWz1yn+Yu9Rytq0cPsSKe1U6d5P+DtIVqN+Yjyy4LE/vP3bL2m01N7Tt
 M5Vef6NwipPKXuaSawyy0I1GV8THBRCe+By13nwJPOqEqgbtK+Nf9FNHMx+dYdDwtake
 uumuUA0iL66N3Bq6QdP56HXhplFRRpl0vtZGLyL6xFNdR90CA+9n73A/Sd0TasWBGW3e
 KR2DHS1H5qQQc/yN9wlLj7zm/tABddgH9frYcf3RDi4MzBYLEVQhtQBuUhbDCGipezqf
 1Qeg==
X-Gm-Message-State: AFqh2koLsJXqYkCbiExy3yZKlwYQcBFSPGDglpmoHuPAT2PYAFIEJt5Z
 eAUGUqUPTuyl1UcwouFqmf5IKg==
X-Google-Smtp-Source: 
 AMrXdXv/kQ382kei19XkFGcLWJ7aPzUvRJJUH21UqXnq9L9JaCiqe3BGGU+X8uqZaE8xThhOGcRsJw==
X-Received: by 2002:a05:6a00:26f2:b0:58d:e099:8cb1 with SMTP id
 p50-20020a056a0026f200b0058de0998cb1mr33390172pfw.33.1674804108146;
 Thu, 26 Jan 2023 23:21:48 -0800 (PST)
Received: from localhost.localdomain
 ([2405:201:c00a:a238:cba6:5f80:3849:2165])
 by smtp.gmail.com with ESMTPSA id
 s23-20020a056a00179700b0058a0e61136asm1974903pfg.66.2023.01.26.23.21.40
 (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
 Thu, 26 Jan 2023 23:21:47 -0800 (PST)
From: Jagan Teki <jagan@edgeble.ai>
To: Tom Rini <trini@konsulko.com>, Kever Yang <kever.yang@rock-chips.com>,
 Simon Glass <sjg@chromium.org>, Philipp Tomsich <philipp.tomsich@vrull.eu>
Cc: u-boot@lists.denx.de, Jagan Teki <jagan@edgeble.ai>,
 Quentin Schulz <quentin.schulz@theobroma-systems.com>
Subject: [PATCH] Revert "rockchip: Only call binman when TPL available"
Date: Fri, 27 Jan 2023 12:51:33 +0530
Message-Id: <20230127072133.852818-1-jagan@edgeble.ai>
X-Mailer: git-send-email 2.25.1
MIME-Version: 1.0
X-BeenThere: u-boot@lists.denx.de
X-Mailman-Version: 2.1.39
Precedence: list
List-Id: U-Boot discussion <u-boot.lists.denx.de>
List-Unsubscribe: <https://lists.denx.de/options/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=unsubscribe>
List-Archive: <https://lists.denx.de/pipermail/u-boot/>
List-Post: <mailto:u-boot@lists.denx.de>
List-Help: <mailto:u-boot-request@lists.denx.de?subject=help>
List-Subscribe: <https://lists.denx.de/listinfo/u-boot>,
 <mailto:u-boot-request@lists.denx.de?subject=subscribe>
Errors-To: u-boot-bounces@lists.denx.de
Sender: "U-Boot" <u-boot-bounces@lists.denx.de>
X-Virus-Scanned: clamav-milter 0.103.6 at phobos.denx.de
X-Virus-Status: Clean

This reverts commit f5315dd6290a588434e4f79bfd2886bb7df9210d.

[why]
TPL is not mandatory for not all Rockchip SoCs, some SoCs like
RK356x, and RK3588 still use mainline u-boot without TPL as
their ddr init programs are accessed via binaries provided by
Rockchip instead of ddr source code.

Marking TPL build makes it not able to build u-boot.itb on
RK356x targets so revert this so that it can build an SPL build
that would support all across Rockchip platforms.

Suggested-by: Quentin Schulz <quentin.schulz@theobroma-systems.com>
Signed-off-by: Jagan Teki <jagan@edgeble.ai>
Tested-by: Anand Moon <linux.amoon@gmail.com>  # CM3
---
 arch/arm/dts/rockchip-u-boot.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/dts/rockchip-u-boot.dtsi b/arch/arm/dts/rockchip-u-boot.dtsi
index 234fc5df43..6d1fd7769e 100644
--- a/arch/arm/dts/rockchip-u-boot.dtsi
+++ b/arch/arm/dts/rockchip-u-boot.dtsi
@@ -11,7 +11,7 @@
 	};
 };
 
-#ifdef CONFIG_TPL
+#ifdef CONFIG_SPL
 &binman {
 	simple-bin {
 		filename = "u-boot-rockchip.bin";
